exploration/04_trigger_exploration.R

```diff
@@ -0,0 +1,885 @@
+    ndvi_cum_anom = ndvi_cum / ndvi_cum_avg
+  ) |> 
+  dplyr$group_by(
+    Month,
+    Dekad
+  ) |> 
+  dplyr$arrange(
+    ndvi_cum,
+    .by_group = TRUE
+  ) |> 
+  dplyr$slice(
+    1:8 # 1 in 5 year RP
+  ) |> 
+  dplyr$ungroup() |> 
+  dplyr$transmute(
+    indicator = "NDVI",
+    month = as.numeric(Month),
+    dekad = as.numeric(Dekad),
+    year = Year
+  )
+
+# since 2000
+NDVI <- df_ndvi |>
+  dplyr$filter(
+    Month %in% c("03", "04", "05"),
+    Year >= 2000
+  ) |> 
+  dplyr$group_by(
+    Year
+  ) |> 
+  dplyr$mutate(
+    ndvi_cum = cumsum(Data),
+    ndvi_cum_avg = cumsum(Data_long_term_Average),
+    ndvi_cum_anom = ndvi_cum / ndvi_cum_avg
+  ) |> 
+  dplyr$group_by(
+    Month,
+    Dekad
+  ) |> 
+  dplyr$arrange(
+    ndvi_cum,
+    .by_group = TRUE
+  ) |> 
+  dplyr$slice(
+    1:5
+  ) |> 
+  dplyr$ungroup() |> 
+  dplyr$transmute(
+    indicator = "NDVI",
+    month = as.numeric(Month),
+    dekad = as.numeric(Dekad),
+    year = Year
+  ) |> 
+  dplyr$filter(
+    month == 5,
+    dekad == 3
+  )
+
+df_chirps_wy <- df_chirps |> 
+  dplyr$mutate(
+    month = lubridate$month(date),
+    year = lubridate$year(date)
+  ) |> 
+  dplyr$filter(
+    month %in% 1:5,
+    year >= 1984
+  ) |> 
+  dplyr$group_by(
+    year
+  ) |> 
+  dplyr$mutate(
+    precipitation_cum = cumsum(value)
+  ) |>
+  dplyr$group_by(
+    month
+  ) |> 
+  dplyr$arrange(
+    precipitation_cum,
+    .by_group = TRUE
+  ) |> 
+  dplyr$slice(
+    1:8 # 1 in 5 year RP
+  ) |> 
+  dplyr$ungroup() |> 
+  dplyr$transmute(
+    indicator = "CHIRPS",
+    month,
+    year
+  )
+
+df_chirps_wy_2000 <- df_chirps |> 
+  dplyr$mutate(
+    month = lubridate$month(date),
+    year = lubridate$year(date)
+  ) |> 
+  dplyr$filter(
+    month %in% 1:5,
+    year >= 2000
+  ) |> 
+  dplyr$group_by(
+    year
+  ) |> 
+  dplyr$mutate(
+    precipitation_cum = cumsum(value)
+  ) |>
+  dplyr$group_by(
+    month
+  ) |> 
+  dplyr$arrange(
+    precipitation_cum,
+    .by_group = TRUE
+  ) |> 
+  dplyr$slice(
+    1:5
+  ) |> 
+  dplyr$ungroup() |> 
+  dplyr$transmute(
+    indicator = "CHIRPS",
+    month,
+    year
+  )
+
+# look at CHIRPS just across MAM
+
+df_chirps_mam_wy <- df_chirps |> 
+  dplyr$mutate(
+    month = lubridate$month(date),
+    year = lubridate$year(date)
+  ) |> 
+  dplyr$filter(
+    month %in% 3:5,
+    year >= 1984
+  ) |> 
+  dplyr$group_by(
+    year
+  ) |> 
+  dplyr$mutate(
+    precipitation_cum = cumsum(value)
+  ) |>
+  dplyr$group_by(
+    month
+  ) |> 
+  dplyr$arrange(
+    precipitation_cum,
+    .by_group = TRUE
+  ) |> 
+  dplyr$slice(
+    1:8 # 1 in 5 year RP
+  ) |> 
+  dplyr$ungroup() |> 
+  dplyr$transmute(
+    indicator = "CHIRPS",
+    month,
+    year
+  )
+
+df_emdat_wy <- df_emdat |> 
+  dplyr$transmute(
+    indicator = "EMDAT",
+    year = c(2000, 2001, 2002, 2007, 2008, 2011, 2018, 2019, 2021, 2022, 2023),
+    faryab = stringr$str_detect(`Location (Affected provinces)`, "Faryab")
+  ) 
```

Comment:
this seems to have been changed in correctly since original commit on this branch